src/core/annotation.js

```diff
@@ -4340,19 +4359,22 @@ class InkAnnotation extends MarkupAnnotation {
   }
 
   static createNewDict(annotation, xref, { apRef, ap }) {
-    const { color, opacity, paths, rect, rotation, thickness } = annotation;
+    const { color, opacity, paths, outlines, rect, rotation, thickness } =
+      annotation;
     const ink = new Dict(xref);
     ink.set("Type", Name.get("Annot"));
     ink.set("Subtype", Name.get("Ink"));
     ink.set("CreationDate", `D:${getModificationDate()}`);
     ink.set("Rect", rect);
-    ink.set(
-      "InkList",
-      paths.map(p => p.points)
-    );
+    ink.set("InkList", outlines?.points || paths.map(p => p.points));
     ink.set("F", 4);
     ink.set("Rotate", rotation);
 
+    if (outlines) {
+      // Free highlight.
+      ink.set("IT", Name.get("InkHighlight"));
+    }
+
```

Comment:
Looking at the PDF specification, I cannot find any mention of `IT` being supported for InkAnnotations; please see https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf#page=400&zoom=auto,0,250:
> (Optional; PDF 1.6) A name describing the intent of the markup
annotation. Intents allow conforming readers to distinguish between
different uses and behaviors of a single markup annotation type. If this
entry is not present or its value is the same as the annotation type, the
annotation shall have no explicit intent and should behave in a generic
manner in a conforming reader.
>
> Free text annotations (Table 174), line annotations (Table 175), polygon
annotations (Table 178), and (PDF 1.7) polyline annotations (Table 178)
have defined intents, whose values are enumerated in the corresponding
tables.

Furthermore, searching for the string "InkHighlight" in the PDF specification produces exactly zero matches. What's the origin of this code, and should we actually set this when it doesn't seems to be defined in the specification?

If this is based on real-world behaviour, e.g. PDFs generated by Adobe software, then please add a comment to that effect to aid future readers!

Reply:
I'll add a comment.
FYI, I found that when using the free highlighting in Edge and when opening the saved pdf in Acrobat  there is something indicating that the ink annotation has been used for highlighting (`surbrillance` in French):
![image](https://github.com/mozilla/pdf.js/assets/5641725/12ef0db5-328f-4dc8-9ac6-2c470dc4ea3c)